### DISCUSSION TITLE
List with dynamic columns (flex)

### DISCUSSION THREAD

**Discussion by charkour** [TanStack/virtual#446] [Q&A] [answered]
Hey all, 

I discovered a cool feature from `react-virtual` that I'd like to port over to tanstack virtual. I've tried a couple of ideas but wasn't sucessful.

Is it possible to make a virtual list with tanstack virtual with a dynamic number of columns? As you resize the screen to make it larger, I'd like there to be more columns. 

Here's a [demo](https://plnkr.co/edit/zjCwNeRZ7XtmFp1PDBsc?p=preview&preview) of what I'd like (and here's the [context](https://stackoverflow.com/questions/46368305/how-to-make-a-list-grid-with-a-responsive-number-of-columns-using-react-virtuali/46392258#46392258)).

Is this possible with tanstack virtual? Thank you!

**Comment by piecyk** [comment, marked as answer]
Hi @charkour, yes, following same approach like @bvaughn did. Checkout this example using window scroll https://codesandbox.io/s/recursing-archimedes-0ngxmg?file=/src/main.tsx

**Reply by charkour** [reply]
Thank you! This worked well and helped me to fix the remaining bugs in my implementation.

This example is really helpful! I could be helpful to add it to the documentation on the examples. Anyways, thanks agin.
